packages/astro/src/types/public/integrations.ts

```diff
@@ -248,10 +245,41 @@ export interface AstroIntegration {
  */
 export type IntegrationRouteData = Omit<
 	RouteData,
-	'isIndex' | 'fallbackRoutes' | 'redirectRoute'
+	'isIndex' | 'fallbackRoutes' | 'redirectRoute' | 'origin'
 > & {
 	/**
 	 * {@link RouteData.redirectRoute}
 	 */
 	redirectRoute?: IntegrationRouteData;
 };
+
+export interface IntegrationResolvedRoute {
```

Comment:
What's the need to create yet another interface? I know that Bjorn expressed a desire to do so. However, the previous PR didn't show an explanation, and even this PR didn't. 

I am fine with any changes, as long as there's an explanation

Reply:
Since we have the opportunity to redesign this API, I think it also makes sense to have a dedicated type. That means we can rename properties the way we want, without being constrainted by the internal naming